File: backend/src/main/java/top/vuhe/entity/UserRole.java
package top.vuhe.entity;

import org.springframework.security.core.GrantedAuthority;

/**
 * @author zhuhe
 */
public enum UserRole implements GrantedAuthority {
    // 管理员
    Admin,
    // 普通用户
    User;

    @Override
    public String getAuthority() {
        return "ROLE_" + toString();
    }
}

File: backend-kt/src/main/kotlin/top/vuhe/sw/entity/auth/UserRole.kt
package top.vuhe.sw.entity.auth

import org.springframework.security.core.GrantedAuthority

/**
 * 用户权限角色
 */
enum class UserRole : GrantedAuthority {
    // 管理员
    Admin,

    // 普通用户
    User;

    override fun getAuthority(): String? {
        return "ROLE_" + toString()
    }
}